kettle-plugins/hadoop-cluster/ui/pom.xml

```diff
@@ -123,6 +123,14 @@
       <scope>test</scope>
     </dependency>
 
+    <!-- This dependency is transitive and should not be marked with dependency="true" because it has only client
+    side code that wouldn't be triggered -->
+    <dependency>
+      <groupId>org.pentaho.di.plugins</groupId>
+      <artifactId>core-ui</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+
```

Comment:
The feature `pentaho-hadoop-cluster-ui` isn't generated here. In fact it is written manually at `assemblies/features/src/main/feature/feature.xml`. You must add the bundle directly there.

You can keep this dependency here, but without the comment and with no version (already defined in its parent pom).